index.js

```diff
@@ -29,6 +29,20 @@ class P2Plex extends EventEmitter {
     if (listenSelf) this.swarm.join(this.publicKey, { announce: true, lookup: false })
   }
 
+  get keyPair () {
+    let { publicKey, secretKey } = this.opts.keyPair || noisePeer.keygen()
```

Comment:
I think it'd be good to move this initialization into the constructor, otherwise we could be calling `noisePeer.keygen()` ever time `this.keypair` is invoked (every time `this.publicKey` is invoked too)

Reply:
@RangerMauve ah good call. Adjusted!